cvxpy/atoms/quad_form.py

```diff
@@ -122,7 +122,7 @@ def _grad(self, values):
         return [sp.csc_matrix(D.ravel(order="F")).T]
 
     def shape_from_args(self) -> Tuple[int, ...]:
-        return tuple() if self.args[0].ndim == 0 else (1, 1)
+        return tuple() if self.args[0].ndim == 0 else ()
```

Comment:
```suggestion
        return tuple()
```
`tuple()` is the same as `()`. 
Also because of the second part of this validation on the inputs above
https://github.com/cvxpy/cvxpy/blob/bf4634cc6c9a35ec31b70c8df81e8aa9759aef55/cvxpy/atoms/quad_form.py#L54-L55
we can never have `self.args[0].ndim == 0`, so this `if` case appears to be redundant.

Reply:
@phschiele thanks a lot for the suggestion! Adjusted here: [Remove redundant if condition](https://github.com/cvxpy/cvxpy/pull/2073/commits/9c119a5bf1461aa1405910f529c6f17495e59cce) and added assertion statement in tests: [Add assert statement in tests](https://github.com/cvxpy/cvxpy/pull/2073/commits/b0c28015cdd11da47fab399aed261518b9c44b1a)